File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeTwoLists(ListNode* list1, ListNode* list2) {
        ListNode* pt1, * pt2;
        pt1 = list1;
        pt2 = list2;

        ListNode* dummy = new ListNode();
        ListNode* pt = dummy;
        while (pt1 != nullptr && pt2 != nullptr) {
            if (pt1->val <= pt2->val) {
                pt->next = pt1;
                pt1 = pt1->next;
                pt = pt->next;
            } else {
                pt->next = pt2;
                pt2 = pt2->next;
                pt = pt->next;
            }
        }
        while (pt1 != nullptr) {
            pt->next = pt1;
            pt1 = pt1->next;
            pt = pt->next;
        }
        while (pt2 != nullptr) {
            pt->next = pt2;
            pt2 = pt2->next;
            pt = pt->next;
        }
        return dummy->next;
    }
};

File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def mergeTwoLists(self, list1, list2):
        """
        :type list1: Optional[ListNode]
        :type list2: Optional[ListNode]
        :rtype: Optional[ListNode]
        """
        if list1 == None:
            return list2
        elif list2 == None:
            return list1

        result = []

        dummyHead = ListNode()
        prev = dummyHead
        
        while (list1 != None and list2 != None):
            if list1.val <= list2.val:
                prev.next = list1
                list1 = list1.next
            else:
                prev.next = list2
                list2 = list2.next
            prev = prev.next

        while (list1 != None):
            prev.next = list1
            list1 = list1.next
            prev = prev.next
        while (list2 != None):
            prev.next = list2
            list2 = list2.next
            prev = prev.next
        return dummyHead.next